qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
If I understand correctly, you are suggesting that we remove all the
prep templates from every study, then re-create the same prep templates
(at this point the new QIIME mapping file will be re-created), then we
reprocess the raw data and then we reprocess the OTU table? I think this
is a lot of unneeded work as the problem is only present in the QIIME
mapping files which are not used for processing, the minimal mapping
file is used for processing and this problem is not propagated there:
https://github.com/biocore/qiita/blob/6614f27a540a6ee652b5e966979c06575056903c/qiita_ware/processing_pipeline.py#L23-73

@josenavas can you correct me if what I'm assuming about the mapping
files is wrong.


On (Feb-10-15| 3:42), Antonio Gonzalez wrote:
> > +from qiita_db.util import get_mountpoint
> > +from qiita_db.sql_connection import SQLConnectionHandler
> > +from qiita_db.metadata_template import PrepTemplate
> > +
> > +conn_handler = SQLConnectionHandler()
> > +
> > +_id, fp_base = get_mountpoint('templates')[0]
> > +
> > +for prep_template_id in conn_handler.execute_fetchall(
> > +        "SELECT prep_template_id FROM qiita.prep_template"):
> > +    prep_template_id = prep_template_id[0]
> > +    pt = PrepTemplate(prep_template_id)
> > +    study_id = pt.study_id
> > +
> > +    for _, fpt in pt.get_filepaths():
> > +        pt.create_qiime_mapping_file(fpt)
> 
> One more pro about reprocessing is that the copied sample/prep template has the original casing. You should also be aware that most of our methods will prepend the name of the files with the study or prep ids so if you are using one of this scripts you should remove those prefixes. 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/829/files#r24406119